uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/absence-tab/components/absence-tab-work-patterns.js

```diff
@@ -10,22 +12,129 @@ define([
       return settings.pathTpl + 'components/absence-tab-work-patterns.html';
     }],
     controllerAs: 'workpatterns',
-    controller: ['$log', '$rootElement', '$uibModal', 'settings', controller]
+    controller: [
+      '$log', '$q', '$rootElement', '$uibModal', 'dialog',
+      'settings', 'OptionGroup', 'WorkPattern', controller]
   });
 
-  function controller($log, $rootElement, $uibModal, settings) {
+  function controller ($log, $q, $rootElement, $uibModal, dialog, settings, OptionGroup, WorkPattern) {
     $log.debug('Component: absence-tab-work-patterns');
 
-    var vm = {};
+    var changeReasons = [];
+    var vm = Object.create(this);
+
+    vm.customWorkpattern = [];
+    vm.defaultWorkPattern = null;
+    vm.loading = true;
+
+    (function init () {
+      refresh(loadJobContractRevisionChangeReasons());
+    })();
+
+    /**
+     * Shows a delete work pattern confirm dialog before deleting the contact work pattern
+     * @param {string} contactWorkPatternID
+     */
+    vm.deleteWorkPattern = function (contactWorkPatternID) {
+      dialog.open({
+        title: 'Confirm Cancellation?',
+        copyCancel: 'Cancel',
+        copyConfirm: 'Confirm',
+        classConfirm: 'btn-danger',
+        msg: 'This cannot be undone',
+        onConfirm: function () {
+          WorkPattern.unassignWorkPattern(contactWorkPatternID)
+            .then(function () {
+              refresh();
+            });
+        }
+      });
+    };
 
     // TODO -This is temporary to open the modal, test cases are pending
```

Comment:
Please change `TODO` to `@TODO`. Could you please also improve this comment by adding a bit more info on why is it temporary and when does it need to be refactored?